### PR TITLE
feat: update header to match refined mocks (#569)

### LINKED ISSUE #569: Update header to match refined mocks
See ticket https://github.com/clevercanary/cc-design/issues/3

### PR DESCRIPTION
**Pull request by frano-m**
### Ticket
Closes #569.

### Reviewers
@NoopDog

### Changes
- Modified header to match refined [mocks](https://www.figma.com/file/i4uZsYkKAML9ivi2TIo6ug/explore-dashboard?node-id=2718%3A27921&t=3SXJbJQKEb9lBN0c-4). Header navigation menu buttons "More" and "Follow Us" show between 1280px and 1440px.
  - Modified header configuration field to take a navigation menu item.
  - Modified social interface and associated configuration by extending the interface with a displayable social `label` value.
  - Refactored social configuration into `constants.ts` file.
  - Modified header component and all associated header components breakpoint from desktop to small desktop (1280px) as per mock. (Added new intermediate breakpoint `desktopSm`.)
  - 

### Definition of Done (from ticket)
- Header matches refined [mocks](https://www.figma.com/file/i4uZsYkKAML9ivi2TIo6ug/explore-dashboard?node-id=2718%3A27921&t=3SXJbJQKEb9lBN0c-4).

### CODE REVIEW

File: explorer/app/components/Layout/components/Footer/footer.stories.tsx [hunk 1]
```diff
@@ -67,19 +68,19 @@ AnvilFooter.args = {
     ],
     socials: [
       {
-        type: "discourse",
+        ...SOCIAL.DISCOURSE,
```
Comment (frano-m) [inline explorer/app/components/Layout/components/Footer/footer.stories.tsx:71]: Footer and Header stories' socials updated with `label` via the new `SOCIAL` const.

File: explorer/app/components/Layout/components/Header/components/Content/content.tsx [hunk 2]
```diff
@@ -4,20 +4,20 @@ import { HEADER_HEIGHT } from "../../header";
 
 interface Props {
   children: ReactNode | ReactNode[];
-  desktop: boolean;
+  desktopSm: boolean;
```
Comment (frano-m) [inline explorer/app/components/Layout/components/Header/components/Content/content.tsx:7]: All changes from `desktop` to `desktopSm`, here and elsewhere, are a change in breakpoint media query for the header.

File: explorer/app/components/Layout/components/Header/header.tsx [hunk 3]
```diff
@@ -35,70 +36,76 @@ export const Header = ({ header }: Props): JSX.Element => {
...
-          <Socials
-            buttonSize={desktop ? "small" : "xlarge"}
-            socials={socials}
-            sx={{
-              gap: desktop ? 2 : 4,
-              px: desktop ? undefined : 4,
-              py: desktop ? undefined : 2,
-            }}
-          />
+          {!onlySmDesktop && (
```
Comment (frano-m) [inline explorer/app/components/Layout/components/Header/header.tsx:89]: We render the usual social icons for mobile, tablet, and desktop and hide them for small desktop when they will be part of the "Follow Us" menu.

File: explorer/app/components/common/Socials/socials.tsx [hunk 4]
```diff
@@ -6,7 +6,16 @@ import { IconName } from "../CustomIcon/common/iconSvgPathShapes";
 import { CustomIcon } from "../CustomIcon/customIcon";
 import { IconButtonSocials } from "../IconButton/iconButton.styles";
 
+export const SOCIAL: Record<string, Omit<Social, "url">> = {
```
Comment (frano-m) [inline explorer/app/components/common/Socials/socials.tsx:9]: New constant to define social label and corresponding type for building the `icon`.

File: explorer/app/hooks/useBreakpointHelper.ts [hunk 5]
```diff
@@ -7,17 +7,19 @@ import { Breakpoint, Theme, useMediaQuery } from "@mui/material";
 
 export const enum BREAKPOINT {
   DESKTOP = "desktop",
+  DESKTOP_SM = "desktopSm",
```
Comment (frano-m) [inline explorer/app/hooks/useBreakpointHelper.ts:10]: New breakpoint added to the media query and theme `desktopSm`.

**Review by NoopDog — APPROVED** [review]

**Pull request merged**